diff --git a/xbmc/application/Application.cpp b/xbmc/application/Application.cpp
index 0765cf3723..09e0e467ff 100644
--- a/xbmc/application/Application.cpp
+++ b/xbmc/application/Application.cpp
@@ -2942,7 +2942,8 @@ bool CApplication::OnMessage(CGUIMessage& message)
       return true;
 
   case GUI_MSG_PLAYBACK_ERROR:
-    HELPERS::ShowOKDialogText(CVariant{16026}, CVariant{16027});
+    // hide unsupportable message because it's annoying.  Thanks to Batocera for the patch.
+    //HELPERS::ShowOKDialogText(CVariant{16026}, CVariant{16027});
     return true;
 
   case GUI_MSG_PLAYLISTPLAYER_STARTED:
@@ -3077,6 +3078,10 @@ void CApplication::ConfigureAndEnableAddons()
 
       if (isConfigureAddonsAtStartupEnabled)
       {
+        if (addon->Type() == AddonType::PVRDLL) // skip and do not initialized PVR plugins
+        {
+          continue;
+        }
         if (HELPERS::ShowYesNoDialogLines(CVariant{24039}, // Disabled add-ons
                                           CVariant{24059}, // Would you like to enable this add-on?
                                           CVariant{addon->Name()}) == DialogResponse::CHOICE_YES)
